src/main/java/com/team766/controllers/CanSparkMaxSmartMotionRotationalPIDController.java

```diff
@@ -0,0 +1,218 @@
+package com.team766.controllers;
+
+import com.team766.config.ConfigFileReader;
+import com.team766.hal.RobotProvider;
+import com.team766.library.SetValueProvider;
+import com.team766.library.SettableValueProvider;
+import com.team766.library.ValueProvider;
+import com.team766.logging.Category;
+import com.team766.logging.Logger;
+import com.team766.logging.Severity;
+import edu.wpi.first.math.MathUtil;
+import com.revrobotics.SparkMaxAbsoluteEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.revrobotics.SparkMaxAbsoluteEncoder.Type;
+import java.io.IOError;
+import javax.swing.text.DefaultStyledDocument.ElementSpec;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.revrobotics.CANSparkMax;
+import com.team766.framework.Mechanism;
+import com.team766.hal.MotorController;
+import com.team766.library.RateLimiter;
+
+
+public class CanSparkMaxSmartMotionRotationalPIDController{
+	
+	// The attributes of the class include references to the motor controller, SparkMax controller, PID controller, and absolute encoder
+	private MotorController mc;
+	private CANSparkMax csm;
+	private SparkMaxPIDController pid;
+	private SparkMaxAbsoluteEncoder abs;
+	
+	//PID Related Variables
+	private static double deadzone = 0; 
+	private static double setPointPosition = 0;
+	private static double comboOfTimesInsideDeadzone = 0;
+	private static double minPos = 0;
+	private static double maxPos = 0;
+
+
+	//Precision variables
+	private double degreesToEncoderUnitsRatio = 0;
+	private double encoderUnitsPerOneAbsolute = 0;
+
+	//antigrav coefficient
+	private static double antiGravK;
+	//enum for which state the PID is in
+	public enum PIDSTATE{
+		PID,
+		OFF,
+		ANTIGRAV
+	}
+	//the state of the PID
+	private PIDSTATE theState = PIDSTATE.OFF;
+	
+	//constructor for the class not using an absolute encoder for kDutyCycle
+	
+	public CanSparkMaxSmartMotionRotationalPIDController(String configName, double absEncoderOffset, double absEncoderOffsetForZeroEncoderUnits, OffsetPoint first, OffsetPoint second, double ratio){
```

Comment:
it seems like we could consolidate the parameters to just `String configName, OffsetPoint first, OffsetPoint second` if you added another field to `OffsetPoint` that gives the position in degrees (so each OffsetPoint would have its position in abs encoder units, motor encoder units, and degrees). the other values should be calculable from those.

though maybe name it something more generic than "degrees" so that it's usable for the non-rotational version as well